### PR TITLE
Uhm 4211 scripts mh laptops

### PR DESCRIPTION
**Pull request by jmbabazi**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,20 +25,28 @@ git clone https://github.com/PIH/malawi-child-puppet.git mental-health
...
 ```
-scp USER_NAME@amigo.pih-emr.org://home/backups/binaries/mental-health/hieradata/common.yaml .
+cd mental-health/hieradata
+scp USER_NAME@amigo.pih-emr.org:/home/backups/binaries/mental-health/hieradata/common.yaml .
 ```
 
-3) Also from amigo.pih-emr.org download "/home/backups/binaries/mental-health/bin/*" (you can fetch an entire directory structure using sftp using "get -r", or you may want to zip up the entire bin directory first)
+3) Also from amigo.pih-emr.org download "/home/backups/binaries/mental-health/bin/*" and the configuration files "/home/backups/binaries/mental-health/openmrs-config-zl.zip"  (you can fetch an entire directory structure using sftp using "get -r", or you may want to zip up the entire bin directory first)
     Create a new directory on the flash drive, "/mental-health/bin" and copy over the contents of the directory downloaded in the previous step
 ```
```
Comment (mogoodrich) [inline README.md]: I think the configuration files should "live" in the same location as the PIH EMR debian package. (That is on the bamboo server in "/home/emradmin/mental-health/deployment")... so I'd put this in "/home/emradmin/mental-health/deployment/configuration" and have users fetch it from there.

Alternatively, we could move both PIH EMR debian package and configuration zip to amigo and have the end users fetch everything from there.

File: hieradata/common.yaml [hunk 2]
```diff
@@ -1,35 +1,34 @@
...
-sync_admin_email: [SYNC_ADMN_EMAIL]
-pih_home: 'c:\pih'
-puppet_install_home: 'c:\malawi-child-puppet'
+windows_openmrs_user: PIH
+
+pih_home: 'c:\openmrs'
 win_startup_menu: 'c:\ProgramData\Microsoft\Windows\Start Menu\Programs'
 mysql_root_user: root
-mysql_root_password: [MYSQL_ROOT_PASSWORD]
+mysql_root_password: omrs123
```
Comment (mogoodrich) [inline hieradata/common.yaml]: We shouldn't have the root password in plaintext here... I think the idea was we were just having the implementor copy it in manually when deploygng?

File: hieradata/common.yaml [hunk 3]
```diff
@@ -1,35 +1,34 @@
...
-openmrs_db_user: [OPENMRS_DB_USER]
-openmrs_db_password: [OPENMRS_DB_PASSWORD]
-download_db: true
-ssh_parent_address: 172.16.1.4
-parent_mysql_db_password: [PARENT_MYSQL_PASSWORD]
-ssh_user: [SSH_USER]
+openmrs_db_user: openmrs
+openmrs_db_password: omrs123
+openmrs_scheduler_user: admin
+openmrs_scheduler_password: SanteMentale123
```
Comment (mogoodrich) [inline hieradata/common.yaml]: Same thing, we  shouldn't have all the passwords in plaintext here...

File: hieradata/common.yaml [hunk 4]
```diff
@@ -1,35 +1,34 @@
...
 label_check_for_unsynced_records: 'Check for Unsynced records'
 label_prepare_child_server: 'Prepare Child Server'
 label_register_child_with_parent: 'Register Child With Parent'
 label_remove_changelock: 'Remove Changelock'
 label_unsynced_changes: 'Remove Unsynced Changes'
 label_shutdown_openmrs: 'Shutdown OpenMRS'
 label_start_openmrs: 'Start OpenMRS'
-label_refresh_ic3_screening_app: 'Refresh IC3 App'
-label_download_modules_from_test: 'Download OpenMRS modules from test server'
-label_refresh_openmrs_app: 'Refresh OpenMRS App'
```
Comment (mogoodrich) [inline hieradata/common.yaml]: Did we change the way this works?

File: install.bat [hunk 5]
```diff
@@ -12,4 +12,5 @@ del /F /Q modules\openmrs\files\modules\*
 xcopy bin\openmrs\openmrs.sql.zip modules\openmrs\files\ /Y /I
 xcopy bin\openmrs\openmrs.war modules\openmrs\files\ /Y /I
 xcopy bin\openmrs\modules\*.omod modules\openmrs\files\modules\ /Y /I
+xcopy bin\openmrs\openmrs-config-zl.zip modules\openmrs\files\ /Y /I
```
Comment (mogoodrich) [inline install.bat]: Cool, this looks good!

File: modules/openmrs/manifests/init.pp [hunk 6]
```diff
@@ -69,7 +70,25 @@
 		force   => true,
 		recurse => true,
 	} ->
+
+    file { $pih_config_dir:
+    	ensure  => directory,
+    	require => File[$pih_home],
```
Comment (mogoodrich) [inline modules/openmrs/manifests/init.pp]: I think we might need to set "purge", "force", and "recurse" here to true to make sure it cleans out this directory each time?

File: hieradata/common.yaml [hunk 7]
```diff
@@ -1,35 +1,34 @@
 sitename: common
-windows_openmrs_user: openmrs
-sync_parent_name: parent-lisungwi-reporting
-sync_parent_address: 'https://172.16.1.4/openmrs'
-sync_parent_uuid: 'SYNC_PARENT_UUID'
-sync_parent_user_name: '[SYNC_USER_NAME]'
-sync_parent_user_password: '[SYNC_USER_PASSWORD]'
-sync_admin_email: [SYNC_ADMN_EMAIL]
-pih_home: 'c:\pih'
-puppet_install_home: 'c:\malawi-child-puppet'
```
Comment (mogoodrich) [inline hieradata/common.yaml]: Are we still using this script to install the Malawi Child servers as well?  If so, we can't remove this?

File: hieradata/common.yaml [hunk 8]
```diff
@@ -1,35 +1,34 @@
...
+openmrs_db_password: omrs123
+openmrs_scheduler_user: admin
+openmrs_scheduler_password: SanteMentale123
+
+ssh_parent_address: bamboo.pih-emr.org
+ssh_user: emradmin
 ssh_port: 22
-artifacts_url: 'http://bamboo.pih-emr.org/artifacts/'
-artifacts_modules_url: 'http://bamboo.pih-emr.org/malawi-repo/modules.zip'
-ic3_app_artifact: openmrs-pwa-apzu-ic3.tar.gz
```
Comment (mogoodrich) [inline hieradata/common.yaml]: Same question: Are we still using this script to install the Malawi Child servers as well? If so, we can't remove this?

File: hieradata/common.yaml (hunk 2 above)
Comment (jmbabazi) [reply hieradata/common.yaml]: Good catch @mogoodrich, I committed from the vm itself(my bad!). I will delete this but I think we should also emphasize on a user changing the password after copying the common.yaml from the repo

Comment (mogoodrich) [reply hieradata/common.yaml]: Sounds good!

**Pull request merged**